APMrover2/GCS_Mavlink.cpp

```diff
@@ -1143,7 +1148,13 @@ void GCS_MAVLINK_Rover::handleMessage(mavlink_message_t* msg)
                 rover.mode_guided.set_desired_location(target_loc);
             } else if (pos_ignore && !vel_ignore && acc_ignore && yaw_ignore && yaw_rate_ignore) {
                 // consume velocity
-                rover.mode_guided.set_desired_heading_and_speed(target_yaw_cd, target_speed);
+                rover.mode_guided.set_desired_heading_and_speed(target_yaw_cd, target_dir * target_speed);
```

Comment:
This doesn't seem quite correct to me.  I think if we received a message with the frame set to MAV_FRAME_BODY_NED or MAV_FRAME_BODY_OFFSET_NED and a velocity x and y pointing about 100 degrees (i.e. back and right) the result would be that the vehicle would turn 100 degrees and then back away.  It would move in the opposite direction from the direction the vector was pointed in.

This wouldn't be the expected behaviour I think.  The expected behaviour, if a user only provides a vector in the body frame (and no yaw) that the vehicle would turn and drive along that vector.  I don't think that just because the vehicle happens to be currently pointed >90 degrees from this vector that the vehicle should drive backwards.  Certainly it shouldn't point along the vector but then travel in the opposite direction.

Perhaps I'm missing something but from inspection, this is what I think would happen.

Reply:
Once again you are right. I keep focusing on one direction and not the vector ... Damned. That would be to bad to not be able to have reverse drive with setpoint.

Reply:
In fact, I just miss a check. The rover should be able to use direction only if a yaw rate is pass, saying that a rate is wanted and not the vector direction. like in local_ned